src/providers/Search/Search.provider.jsx

```diff
@@ -0,0 +1,24 @@
+import React, { useState, useContext } from 'react';
+
+const SearchContext = React.createContext(null);
+
+function useSearch() {
+  const context = useContext(SearchContext);
+  if (!context) {
+    throw new Error(`Can't use "useSearch" without an SearchContext!`);
```

Comment:
I believe that you meant `SearchProvider`, the only way that the condition `!context` resolves to `true` only when you use the hook `useSearch` on a component that is not a child from a SearchProvider component.

You could actually see the error being thrown if you remove from `index` the `SearchProvider`